php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1479 @@
+  }
+
+  public VariableIdentifierTree ENCAPSULATED_VARIABLE_IDENTIFIER() {
+    return b.<VariableIdentifierTree>nonterminal(PHPLexicalGrammar.ENCAPS_VAR_IDENTIFIER).is(
+      // variable identifiers encapsulated into strings literal does not allowed line terminator spacing,
+      // so here using "WHITESPACE" instead of SPACING.
+      f.encapsulatedVariableIdentifier(b.token(PHPLexicalGrammar.WHITESPACES), b.token(PHPLexicalGrammar.VARIABLE_IDENTIFIER)));
+  }
+
+  public ExpressionTree EXPANDABLE_STRING_CHARACTERS() {
+    return b.<ExpandableStringCharactersTree>nonterminal(Kind.EXPANDABLE_STRING_CHARACTERS).is(
+      f.expandableStringCharacters(b.token(PHPLexicalGrammar.STRING_WITH_ENCAPS_VAR_CHARACTERS)));
+  }
+
+  public YieldExpressionTree YIELD_EXPRESSION() {
+    return b.<YieldExpressionTree>nonterminal(Kind.YIELD_EXPRESSION).is(
+      f.yieldExpression(
+        b.token(YIELD),
+        EXPRESSION(),
+        b.optional(f.newTuple1(b.token(DOUBLEARROW), EXPRESSION()))));
+  }
+
+  public ParenthesisedExpressionTree PARENTHESIZED_EXPRESSION() {
+    return b.<ParenthesisedExpressionTree>nonterminal(Kind.PARENTHESISED_EXPRESSION).is(
+      f.parenthesizedExpression(
+        b.token(LPARENTHESIS),
+        b.firstOf(
+          YIELD_EXPRESSION(),
+          EXPRESSION()),
+        b.token(RPARENTHESIS)));
+  }
+
+  public AssignmentExpressionTree LIST_EXPRESSION_ASSIGNMENT() {
+    return b.<AssignmentExpressionTree>nonterminal(PHPLexicalGrammar.LIST_EXPRESSION_ASSIGNMENT).is(
+      f.listExpressionAssignment(LIST_EXPRESSION(), b.token(EQU), EXPRESSION()));
+  }
+
+  public ListExpressionTree LIST_EXPRESSION() {
+    return b.<ListExpressionTree>nonterminal(Kind.LIST_EXPRESSION)
+      .is(f.listExpression(
+          b.token(LIST),
+          b.token(LPARENTHESIS),
+          b.optional(LIST_ELEMENT()),
+          b.zeroOrMore(f.newTuple3(b.token(COMMA), b.optional(LIST_ELEMENT()))),
+          b.token(RPARENTHESIS))
+      );
+  }
+
+  public ExpressionTree LIST_ELEMENT() {
+    return b.<ExpressionTree>nonterminal().is(
+      b.firstOf(
+        MEMBER_EXPRESSION(),
+        LIST_EXPRESSION()));
+  }
+
+  public ComputedVariableTree COMPUTED_VARIABLE_NAME() {
+    return b.<ComputedVariableTree>nonterminal(Kind.COMPUTED_VARIABLE_NAME).is(
+      f.computedVariableName(
+        b.token(LCURLYBRACE),
+        EXPRESSION(),
+        b.token(RCURLYBRACE)));
+  }
+
+  public VariableIdentifierTree VARIABLE_IDENTIFIER() {
+    return b.<VariableIdentifierTree>nonterminal(Kind.VARIABLE_IDENTIFIER).is(
+      f.variableIdentifier(b.token(PHPLexicalGrammar.REGULAR_VAR_IDENTIFIER)));
+  }
+
+  public VariableTree COMPOUND_VARIABLE() {
+    return b.<VariableTree>nonterminal(Kind.COMPOUND_VARIABLE_NAME).is(
+      b.firstOf(
+        VARIABLE_IDENTIFIER(),
+        f.compoundVariable(b.token(DOLLAR_LCURLY), EXPRESSION(), b.token(RCURLYBRACE))));
+  }
+
+  public ExpressionTree VARIABLE_WITHOUT_OBJECTS() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)